ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -523,7 +522,10 @@ private Optional<UInt64> getEarliestAvailabilityWindowSlotBeforeBlock(
     }
     final UInt64 firstAvailabilityWindowSlot =
         spec.computeStartSlotAtEpoch(
-                currentEpoch.minusMinZero(MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS))
+                currentEpoch.minusMinZero(
```

Comment:
An idea to remove these spaghetti could be to move methods like this to ForkChoiceUtil  and we could start having multiple milestone-based implementations.

Reply:
Doesn't look very good passing spec there https://github.com/Consensys/teku/pull/7304/commits/c580aa7f3d94e77783b2bc412b7c942238757248